packages/sanity/src/_internal/cli/util/__tests__/compareStudioDependencyVersions.test.ts

```diff
@@ -0,0 +1,177 @@
+import {beforeEach, describe, expect, it, jest} from '@jest/globals'
+import resolveFrom from 'resolve-from'
+
+import {compareStudioDependencyVersions} from '../compareStudioDependencyVersions'
+import {readPackageJson} from '../readPackageJson'
+
+jest.mock('resolve-from')
+jest.mock('../readPackageJson')
+
+global.fetch = jest.fn() as jest.MockedFunction<typeof fetch>
```

Comment:
I don't feel very comfortable with this approach, since it'll potentially leak into other tests running in the same process (not _quite_ certain how much jest/vitest isolates these, but feels dodgy). Could we make `fetch` an optional argument to `compareStudioDependencyVersions`, so we can inject it?